cpp/src/io/orc/aggregate_orc_metadata.cpp

```diff
@@ -249,17 +249,17 @@ std::vector<metadata::stripe_source_mapping> aggregate_orc_metadata::select_stri
 }
 
 column_hierarchy aggregate_orc_metadata::select_columns(
-  std::vector<std::string> const& column_paths)
+  std::optional<std::vector<std::string>> const& column_paths)
 {
   auto const& pfm = per_file_metadata[0];
 
   column_hierarchy::nesting_map selected_columns;
-  if (column_paths.empty()) {
+  if (not column_paths.has_value()) {
```

Comment:
TIL: I thought we could do:
```c++
if (not column_paths) {...}
```
But the `optional::operator bool()` is `explicit`.

Reply:
That approach works. I just prefer this version, more explicit :)

Reply:
> That approach works.

Wait, really? I thought the [`explicit operator bool()`](https://en.cppreference.com/w/cpp/utility/optional/operator_bool) would disallow it. 

Hmm. I'll check.